solidity-plugin/src/main/java/org/sonarsource/solidity/SyntaxHighlightingVisitor.java

```diff
@@ -24,13 +24,14 @@ public void visitTokens(TokenStream tokenStream) {
       } else if (SoliditySensor.KEYWORD_TYPES.contains(token.getText())) {
         this.highlighting.highlight(token.getLine(), token.getCharPositionInLine(),
           token.getLine(), (token.getCharPositionInLine() + token.getStopIndex() - token.getStartIndex() + 1), TypeOfText.KEYWORD_LIGHT);
-      } else if (token.getType() == 115) {
+      } else if (SolidityParser.typeMatches(token, SolidityParser.StringLiteral)) {
         this.highlighting.highlight(token.getLine(), token.getCharPositionInLine(),
           token.getLine(), (token.getCharPositionInLine() + token.getStopIndex() - token.getStartIndex() + 1), TypeOfText.STRING);
-      } else if (token.getType() >= 96 && token.getType() <= 100) {
+      } else if (SolidityParser.typeMatches(token, 95, 96, 97, 98, 99, 100)) {
         this.highlighting.highlight(token.getLine(), token.getCharPositionInLine(),
           token.getLine(), (token.getCharPositionInLine() + token.getStopIndex() - token.getStartIndex() + 1), TypeOfText.CONSTANT);
       } else {
+        // TODO maybe add more cases
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=19&issues=AWQYrdmJbXXTsx7O9fim&open=AWQYrdmJbXXTsx7O9fim)

[ISSUE_KEY]: AWQYrdmJbXXTsx7O9fim